File: src/www/__test__/testSteps/play/ThereAreNoHighlightedPiles.ts
import { AbstractPostLineStep } from "../AbstractPostLineStep";
import { queryByPileHighlighted } from "./helpers";

export class ThereAreNoHighlightedPiles extends AbstractPostLineStep {
  constructor() {
    super(/There are no highlighted piles/i);
  }

  async runMatch(_: any, match: string[]) {
    const highlighted = queryByPileHighlighted(document.body);
    expect(highlighted).not.toBeInTheDocument();
  }
}

File: src/test/java/com/drpicox/game/testSteps/play/ThereAreNoHighlightedPiles.java
package com.drpicox.game.testSteps.play;

import com.drpicox.game.testPost.reader.PostLine;
import com.drpicox.game.testSteps.AbstractPostLineStep;
import org.springframework.stereotype.Component;

@Component
public class ThereAreNoHighlightedPiles extends AbstractPostLineStep {

    @Override
    protected String getRegex() {
        return "There are no highlighted piles";
    }

    @Override
    protected void run(PostLine line, String[] match) {
        // No implementation required by tests in backend
    }
}
